src/include/daos_srv/pool.h

```diff
@@ -299,4 +300,56 @@ ds_pool_get_version(struct ds_pool *pool)
 	return ver;
 }
 
+/**
+ * Pool service replica clue
+ *
+ * Pool service replica info gathered when glancing at a pool.
+ */
+struct ds_pool_svc_clue {
+	struct rdb_clue	psc_db_clue;
+	uint32_t	psc_map_version;	/**< if 0, empty DB replica */
+};
+
+/** Pool parent directory */
+enum ds_pool_dir {
+	DS_POOL_DIR_NORMAL,
+	DS_POOL_DIR_NEWBORN,
+	DS_POOL_DIR_ZOMBIE
+};
+
+/**
+ * Pool clue
+ *
+ * Pool shard and service replica (if applicable) info gathered when glancing
+ * at a pool. The pc_uuid, pc_dir, and pc_rc fields are always valid; the
+ * pc_svc_clue field is valid only if pc_rc is zero.
+ */
+struct ds_pool_clue {
+	uuid_t				pc_uuid;
+	d_rank_t			pc_rank;
+	enum ds_pool_dir		pc_dir;
+	int				pc_rc;
+	struct ds_pool_svc_clue	       *pc_svc_clue;
+};
+
+void ds_pool_glance(uuid_t uuid, enum ds_pool_dir dir, struct ds_pool_clue *clue);
+void ds_pool_clue_fini(struct ds_pool_clue *clue);
+
+/** Array of ds_pool_clue objects */
+struct ds_pool_clues {
+	struct ds_pool_clue    *pcs_array;
+	int			pcs_len;
+	int			pcs_cap;
+};
+
+/**
+ * If this callback returns 0, the pool with \a uuid will be glanced at;
+ * otherwise, the pool with \a uuid will be skipped.
+ */
+typedef int (*ds_pool_scan_filter_t)(uuid_t uuid, void *arg);
+
+int ds_pool_scan(ds_pool_scan_filter_t filter, void *filter_arg, struct ds_pool_clues *clues_out);
```

Comment:
very minor thought, up to you if you think it's worth making any change. It would be only for a slight enhancement to understanding by the code reader / other developers.

Naming this function ds_pool_clues_scan() or perhaps ds_pool_clues_init() may give a better sense that it is potentially operating over multiple pools, and that the output is a struct ds_pool_clues. "scan" does imply multiple pools already, so the "init" idea may not be so useful. And including ds_pool_clues_ in the function name makes a rough association with the finalization function name ds_pool_clues_fini().

Similarly, at the single pool level, renaming ds_pool_glance to ds_pool_clue_glance() or ds_pool_clue_init() may be similarly helpful,

Reply:
Interestingly, I had an almost identical thought process when writing the code: "glimpse" renamed to "init" renamed to "scan". Let us go with "init" then for symmetry, and put "scan" into the doc of "init".